docs/docs/02-diagram_format.md

```diff
@@ -0,0 +1,376 @@
+| `list`                                | List/array        | [Lists](#lists)                 |
+| `tuple`                               | Tuple             | [Tuples](#tuples)               |
+| `set`                                 | Set               | [Sets](#sets)                   |
+| `dict`                                | Dictionary        | [Dictionaries](#dictionaries)   |
+| `int`, `str`, `bool`, `float`, `None` | Primitive types   | [Primitives](#primitives)       |
+
+---
+
+### Stack Frames
+
+| Attribute | Type                | Required | Description                                                                                                                                                                                                                                        |
+| --------- | ------------------- | -------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `type`    | `string`            | Yes      | Must be `.frame` to indicate a stack frame.                                                                                                                                                                                                        |
+| `id`      | `null`              | Yes      | Must be `null` for stack frames                                                                                                                                                                                                                    |
+| `name`    | `string`            | Yes      | The name of the stack frame (e.g., `__main__`, `foo`)                                                                                                                                                                                              |
+| `value`   | `object`            | Yes      | A mapping of variable names (as keys) to object IDs (as values). These represent the local variables in the scope of the frame. Variable names may be empty strings (`""`) and attribute values may be `null` to render blank boxes, respectively. |
+| `x`, `y`  | `number`            | No       | Optional manual coordinates. Required only if `automation` is disabled.                                                                                                                                                                            |
+| `style`   | `object` or `array` | No       | Custom visual styling.                                                                                                                                                                                                                             |
+
+#### Examples
+
+<details>
+    <summary>**Main Stack Frame**</summary>
```

Comment:
In general for all example titles: just capitalize the first word, and don't capitalize other words